include/stdint.h

```diff
@@ -41,6 +41,59 @@
 #ifndef _MACHINE__STDINT_H_
 #define	_MACHINE__STDINT_H_
 
+/*
+ * Useful integer type names that we can't pick up from the compile-time
+ * environment.
+ */
+typedef char		int8_t;
```

Comment:
Clang (and I believe modern GCC) defines all of these for you. You get macros like `__UINT8_TYPE__`, and even the `__UINT_LEAST16_TYPE__`, `__UINT_FAST32_TYPE__` etc. Given this header is wrong for 32-bit systems I would suggest just using all those; CheriOS would conceivably be highly useful for RV32 CHERI-RISC-V given its utility in embedded systems, once the issue of type bits is addressed.

Reply:
If this just works, seconded.

Reply:
Can we just do `#include_next <stdint.h>` to pick up the compiler-provided one?

Reply:
Yes, that works if we get rid of `-nostdinc`.  Are we going to have `-ibuiltininc` soon on CTSRD-CHERI/llvm-project?

Reply:
Currently it's Darwin-only upstream.

Reply:
Weird for that to be OS-specific.  Would it be easy to add to our frontend?  I feel a little jittery about removing `-nostdinc` from cherios.  It's probably fine for my segregated build environment where the cherios builds can't see the mainline/cheribsd builds and their sysroots, but...

Reply:
Our usual "hack" is `--sysroot=/no/such/path`...

Reply:
And yeah, well, `-nostdinc` is already OS-specific, don't ask me why. There are some really bad decisions in the structuring of Clang's drivers. And no, no reason not to add it to the FreeBSD driver, though it's not strictly required and does diverge from upstream. But when Lawrence adds a CheriOS triple and driver, _that_ driver can always support `-ibuiltininc` :)